server/api/api/endpoints/nuclio.py

```diff
@@ -47,3 +48,24 @@ async def list_api_gateways(
         ),
         auth_info,
     )
+
+
+@router.post("/projects/{project}/nuclio/api-gateways/{gateway}")
+async def create_api_gateway(
```

Comment:
A general question - does the nuclio create-gw API perform create-only (i.e. fails if object already exists) logic or does it do a create-or-replace?
Reason I'm asking is that in MLRun usually we use `POST` for create-only, and `PUT` for create-or-replace, and also the endpoint function would usually be called `store_XXX` in the create-or-replace case.

Reply:
@theSaarco @rokatyy in Nuclio `POST /api_gateways` fails if an api gateway with the same name already exists.

Reply:
I've checked Nuclio's `Update` method and it updates the entity if it already exists. So, replaced `POST` with `PUT`.

Reply:
@rokatyy what @theSaarco meant here is that in mlrun we use `create_...` [POST method] when we only want to create a new object (meaning - fail on existing), and use `store_...` [PUT method] when we want it to "create-or-update" (meaning, create if doens't exist, otherwise update).

Here you did a mix between the two - POST `create_..` but with create-or-update behavior.